UnityProject/Assets/Scripts/Messages/Server/DeviceUpdateLinkerInfoMessage.cs

```diff
@@ -0,0 +1,73 @@
+using System.Collections.Generic;
+using System.Linq;
+using Messages.Server;
+using Mirror;
+using Newtonsoft.Json;
+using Shared.Systems.ObjectConnection;
+using UnityEngine;
+
+public class DeviceUpdateLinkerInfoMessage : ServerMessage<DeviceUpdateLinkerInfoMessage.NetMessage>
+{
+	public struct NetMessage : NetworkMessage
+	{
+		public string ConnectedDevices;
+		public uint MasterID;
+	}
+
+	public override void Process(NetMessage msg)
+	{
+		var Slaves = JsonConvert.DeserializeObject<List<uint>>(msg.ConnectedDevices);
+		InGameDeviceLinker.Instance.MastersData.Clear();
+		var Master = msg.MasterID.NetworkIdentity().GetComponent<IMultitoolMasterable>();
```

Comment:
no fail safe on this?

Reply:
is a server message so, would be the server messing up if this was null

Reply:
there is no chance that you could send the wrong net id over to the server?

Reply:
Would error on the request
```
		public override void Process(NetMessage msg)
		{
			if (IsFromAdmin() == false) return;

			if (msg.Slave is NetId.Empty or NetId.Invalid)
			{
				var Master = msg.Master.NetworkIdentity().GetComponent<IMultitoolMasterable>();
				InGameDeviceLinker.Instance.Refresh();
				var Object = SentByPlayer.GameObject;
				DeviceUpdateLinkerInfoMessage.SendTo(Object, InGameDeviceLinker.Instance.MastersData[msg.MultitoolConnectionType][Master], Master.gameObject);
				return;
			}
			```

Reply:
ah okay then, carry on